libraries/AP_HAL/DSP.cpp

```diff
@@ -164,10 +183,25 @@ uint16_t DSP::step_calc_frequencies(FFTWindowState* fft, uint16_t start_bin, uin
     return fft->_peak_data[CENTER]._bin;
 }
 
+void DSP::update_average_from_sliding_window(FFTWindowState* fft)
+{
+    // copy and scale the new slice
+    float* slice = &fft->_sliding_window[fft->_current_slice * (fft->_bin_count + 1)];
```

Comment:
array index asserts in SITL would be good, needs autotest to cover this code

Reply:
Fixed